src/main/java/org/craftercms/deployer/impl/TargetImpl.java

```diff
@@ -94,12 +95,18 @@ public static String getId(String env, String siteName) {
         return String.format(TARGET_ID_FORMAT, siteName, env);
     }
 
-    public TargetImpl(ZonedDateTime loadDate, String env, String siteName, String localRepoPath,
-                      File configurationFile, HierarchicalConfiguration<ImmutableNode> configuration,
-                      ConfigurableApplicationContext applicationContext, ExecutorService executor,
-                      TaskScheduler scheduler, TargetLifecycleHooksResolver targetLifecycleHooksResolver,
-                      DeploymentPipelineFactory deploymentPipelineFactory, boolean crafterSearchEnabled) {
-        this.loadDate = loadDate;
+    public TargetImpl(
+            @Value("${target.env}") String env,
+            @Value("${target.siteName}") String siteName,
+            @Value("${target.localRepoPath}") String localRepoPath,
+            @Value("${targetConfigFile}") File configurationFile, //TODO: Is it ok to pass via param or remove?
+            @Autowired HierarchicalConfiguration<ImmutableNode> configuration,
+            @Autowired ConfigurableApplicationContext applicationContext,
```

Comment:
I suppose Spring automatically injects here the context where this bean is defined right?

Reply:
Yeah